database-monitoring troubleshooting: add connection test section (#16759)
DataDog/documentation: content/en/database_monitoring/troubleshooting.md

If you think that your setup is correct, but you're not seeing data in your DBM pages, it's possible that your agent is not able to send data to our data collection endpoints. Performing these connection troubleshooting steps from the location where the agent is running will help diagnose any connection issues.

1. Test TCP connectivity on DBM collection endpoints:

```
telnet dbm-metrics-intake.datadoghq.com 443
telnet dbquery-intake.datadoghq.com 443
```

2. Test posting an empty payload with an invalid API key on both DBM endpoints. 
We expect these commands to fail with HTTP code `403: Forbidden`. 

```
curl -vvv -X POST "https://dbm-metrics-intake.datadoghq.com/api/v2/databasequery" \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "DD-API-KEY: NONE" \
-d "[{}]"

curl -vvv -X POST "https://dbquery-intake.datadoghq.com/api/v2/databasequery" \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "DD-API-KEY: NONE" \
-d "[{}]"
```

The responses should contain `{"status":"error","code":403,"errors":["Forbidden"],...}` if requests were successfully sent and a response was received.

> Some common causes of connection failure include proxy setups and firewalls preventing outbound traffic to our endpoints. If you have a proxy or firewall, make sure the IPs addresses for the DBM endpoints are whitelisted. The IP addresses are contained in the APM block defined at https://ip-ranges.datadoghq.com/.

Reviewer: we can also link to the HA proxy config setup docs? https://docs.datadoghq.com/agent/proxy/?tab=linux

Author: good call, will add

Reviewer 2: added!

> Some common causes of connection failure include proxy setups and firewalls preventing outbound traffic to our endpoints. If you have a proxy or firewall, make sure the IPs addresses for the DBM endpoints are whitelisted. The IP addresses are contained in the APM block defined at https://ip-ranges.datadoghq.com/.

Reviewer 2: ```suggestion
Some common causes of connection failure include proxy setups and firewalls, which outbound traffic to Datadog's endpoints. If you have a proxy or firewall, make sure the IPs addresses for the DBM endpoints are allowed. Refer to the APM block in Datadog's [IP addresses](https://ip-ranges.datadoghq.com/).
```
